proposals/1113_ldap-ad-web-ui/README.md

```diff
@@ -66,11 +66,46 @@ custom Username/Password headers, they should be rejected.
 
 The discovery metadata for LDAP/AD IDPS should indicate that they support a flow of `browser_authcode`.
 
-The pinniped cli should default to using the cli-based password flow, but when a tty is unavailable,
-it will open a browser to log in
+The state param should be augmented to include the IDP type as well as the IDP name. The type
+should be included in `UpstreamStateParamData` so that later when we get it back in the callback
+request we can tell which IDP it is referring to. This will require an update to
+`UpstreamStataParamData.FormatVersion`, which would mean that logins in progress at the time of
+upgrade would fail.
+
+The pinniped cli should default to using the cli-based password flow, but when the `--upstream-identity-provider-flow`,
+flag specifies `browser_authcode`, it will open a browser to log in
 instead of prompting for username and password. Some users (for example, IDE plugins for kubernetes)
 may wish to authenticate using the pinniped cli but without access to a terminal.
 
+Here is how the login flow might work:
+1. The supervisor receives an authorization request.
+   1. If the client_id param is not "pinniped-cli", and it includes username and password via the custom headers, reject the request.
+   2. If the request does not include the custom username/password headers, assume we want to use the webpage login.
+   3. Some day we may want to allow specifying the idp name and type as request parameters, but
+      for now we do not have multiple idps.
```

Comment:
Note that the CLI will already send these as parameters, but the server-side code for the authorization endpoint currently ignores them, as you pointed out, since we don't support multiple IDPs yet. Part of implementing multiple IDP support will involve paying attention to these params.